components/cli/pkg/commands/extract_resources.go

```diff
@@ -25,74 +25,69 @@ import (
 	"path/filepath"
 	"time"
 
-	"github.com/cellery-io/sdk/components/cli/pkg/image"
-
+	"github.com/cellery-io/sdk/components/cli/cli"
 	"github.com/cellery-io/sdk/components/cli/pkg/constants"
+	"github.com/cellery-io/sdk/components/cli/pkg/image"
 	"github.com/cellery-io/sdk/components/cli/pkg/util"
 )
 
 // RunExtractResources extracts the cell image zip file and copies the resources folder to the provided path
-func RunExtractResources(cellImage string, outputPath string) {
+func RunExtractResources(cli cli.Cli, cellImage string, outputPath string) error {
 	parsedCellImage, err := image.ParseImageTag(cellImage)
 	if err != nil {
-		util.ExitWithErrorMessage("Error occurred while parsing cell image", err)
+		return fmt.Errorf("error occurred while parsing cell image, %v", err)
 	}
 
-	repoLocation := filepath.Join(util.UserHomeDir(), constants.CELLERY_HOME, "repo", parsedCellImage.Organization,
+	repoLocation := filepath.Join(cli.FileSystem().Repository(), parsedCellImage.Organization,
 		parsedCellImage.ImageName, parsedCellImage.ImageVersion)
 	imageLocation := filepath.Join(repoLocation, parsedCellImage.ImageName+constants.CELL_IMAGE_EXT)
 
 	// Checking if the image is present in the local repo
 	isImagePresent, _ := util.FileExists(imageLocation)
 	if !isImagePresent {
-		util.ExitWithErrorMessage(fmt.Sprintf("Failed to extract resources for image %s", util.Bold(cellImage)),
-			errors.New("Image not Found"))
+		return fmt.Errorf(fmt.Sprintf("failed to extract resources for image %s, %v", util.Bold(cellImage),
+			errors.New("image not Found")))
```

Comment:
We don't have to create another error here now using `errors.New()`. It can be part of the same error message

Previously it was required because we were using `util.ExitWithError()`

Reply:
Updated.